src/lib/provable/core/provable-context.ts

```diff
@@ -71,6 +71,7 @@ function inCompileMode() {
 
 function asProver(f: () => void) {
   if (inCheckedComputation()) {
+    // TODO make this start a "witness block" context
```

Comment:
Can you make an issue and reference it to the comment to tackle later? (if you don't mind, of course) 

I'd like to see more issues in TODO comments like this to make it easier to pick up :D